README.md

```diff
@@ -58,10 +58,70 @@ build/$ ctest . --output-on-failure
 
 The bci.config file is used by our internal validation environment, please just ignore it.
 
+### Build with dilithium support
+
+Dilithium is an optional feature provided by MoCOCrW. To enable the feature,
+replace the CMake invocation with:
+```
+build/$ cmake -DBUILD_TESTING=True -DDILITHIUM_ENABLED=ON ..
+```
+
+Make sure that the adapted version of libdilithium can be found by the linker.
+
+Note, HSM and dilithium support can be enabled independently via CMake arguments.
+
+#### Dilithium Adaptions
+
+It is not possible to take the bare dilithium implementation. The dilithium implementation was
+adapted. A new function for retrieving the public key from a private key was added. This change is
+required to compile MoCOCrW with dilithium support.
+
+There are two PRs created for dilithium. These can be found in dilithium's [github
+repository](https://github.com/pq-crystals/dilithium/).
+
+The first [PR](https://github.com/pq-crystals/dilithium/pull/68) contains the adaptions for
+retreiving the public key from a private key and is **required** in order to compile MoCOCrW with
+dilithium support.
+
+The second [PR](https://github.com/pq-crystals/dilithium/pull/69) improves the CMake file so that
+the static libraries and the header can be installed using CMake. It is recommended to use this PR
+for compiling and installing dilithium.
+
+To get the local copy of libdilithium with above PRs:
+```
+git clone https://github.com/pq-crystals/dilithium.git && cd dilithium
```

Comment:
This looks overly complex. After clone, why can't I check out to  _pub-key-extraction_ and merge _cmake-improvements_ locally?

Reply:
pub-key-extraction is a branch on wusto's fork

Reply:
If you're pulling a pull request like this, hash of the parent commit should match. Wusto made his PRs on a hash mention here (3e9b9f) so that's why branches are switched like that. I don't know if there's a more elegant way to do it.

Reply:
Why do we have this complex instructions here? The README isn't really a how to use git. Just refer to the patch files and be good with it. This is going to be outdated at some point, will then get people to use an old version of libdilithium. 
If you want to use this - its your responsibility to take care of such things. A README is no cooking recipe for software build and packaging.